src/Bynder/Api/Impl/PermanentTokens/RequestHandler.php

```diff
@@ -20,15 +20,23 @@ public function __construct($configuration)
     protected function sendAuthenticatedRequest($requestMethod, $uri, $options = [])
     {
         $request = new \GuzzleHttp\Psr7\Request($requestMethod, $uri, $options);
+        $updatedHeaders = ['headers'=> [
+            'User-Agent' => 'bynder-php-sdk/' . $this->configuration->getSdkVersion()
+            ]
+        ];
+        
+        if(!is_null($options['headers'])) {
+            $updatedHeaders = ['headers' => array_merge($options['headers'],
+                                                        $updatedHeaders['headers']
+                                                        )
+                                ];
+        }
```

Comment:
Can't we do something like this? (not sure about the right syntax for `$options['headers'] || []` here, but at least let it default to `[]` somehow.

```suggestion
        $updatedHeaders = ['headers' => array_merge(
            $options['headers'] || [],
            ['User-Agent' => 'bynder-php-sdk/' . $this->configuration->getSdkVersion()]
        )];
```

Reply:
I don't quite follow. When you say `$options['headers'] || []` I presume when we don't specify `options['headers']` we want it to only contain 'User-Agent' header which is happening right. Because the array_merge is skipped due not is_set

Reply:
Yeah so basically you want the UA header, combined with any other headers if they are there.

So I think my suggestion would have the same result as yours, but is a bit simpler / easier to understand.

Reply:
Ah okay. I used the commit suggestion in conjunction with changing the signature of the method for `options`. This should be much more comprehendible now 😄